inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -126,6 +127,10 @@ CNNNetworkNGraphImpl::CNNNetworkNGraphImpl(const std::shared_ptr<Function>& nGra
     // Add shape infer method for old operations which are not included to opset1, opset2 and opset3
     ::ngraph::op::GenericIE::addExtension(_ngraph_function, std::make_shared<ShapeInfer::BuiltInShapeInferHolder>());
 
+    ngraph::pass::Manager ssr_manager;
+    ssr_manager.register_pass<ngraph::pass::SmartReshape>();
+    ssr_manager.run_passes(_ngraph_function);
```

Comment:
should not this be applied inside IR v10 reader?

Reply:
No, onnx importer path should have it too.

Reply:
Ok, we can add to both readers. I'm just thinking that "we are fixing the original model" which seems to be a part of reader, is not it..?

Reply:
Well, than CNNNetwork creation from handmade nGraph function wouldn't work

Reply:
Do we need to handle such cases? author of handmade ngraph function should provide a "perfect" model.
Originally I supposed we want to fix only MO->IR introduced issues..